src/main/java/frc/robot/subsystems/drivetrain/SwerveModule.java

```diff
@@ -0,0 +1,186 @@
+     */
+    public void setMotorPorts(int[] ports) {
+        driveMotor = new WPI_TalonFX(ports[0]);
+        angleMotor = new WPI_TalonSRX(ports[1]);
+    }
+
+    /**
+     * Configures the inversions for the motors.
+     *
+     * @param inverted is the array of inversions. [drive, angle]
+     */
+    public void configInverted(boolean[] inverted) {
+        driveMotor.setInverted(inverted[0]);
+        angleMotor.setInverted(inverted[1]);
+    }
+
+    /**
+     * Configure sensor phase for module.
+     *
+     * @param sensorPhase is the array of sensor phases. [drive, angle]
+     */
+    public void configSensorPhase(boolean[] sensorPhase) {
+        driveMotor.setSensorPhase(sensorPhase[0]);
+        angleMotor.setSensorPhase(sensorPhase[1]);
+    }
+
+    /**
+     * Sets the postions of the encoders.
+     *
+     * @param positions is the array of positions. [drive, angle]
+     */
+    public void setZeroPosition(int[] positions) {
+        driveMotor.setSelectedSensorPosition(positions[0]);
+        angleMotor.setSelectedSensorPosition(positions[1]);
+    }
+
+    /**
+     * Gets the current angle of the module.
+     *
+     * @return the angle of the module. [deg]
+     */
+    public double getAngle() {
+        return Math.IEEEremainder(
+                angle_unitModel.toUnits(angleMotor.getSelectedSensorPosition()),
+                360
+        );
+    }
+
+    /**
+     * Sets the angle for the module.
+     *
+     * @param reqAngle is the angle to set the module to. [deg]
+     */
+    public void setAngle(double reqAngle) {
+        double currAngle = getAngle();
+
+        currAngle %= 360;
+        currAngle = (currAngle < 0) ? (360 + currAngle) : currAngle;
+
+        reqAngle %= 360;
+        reqAngle = (reqAngle < 0) ? (360 + reqAngle) : reqAngle;
+
+        double error = reqAngle - currAngle;
+        error = (Math.abs(error) > 180) ? (error - (360 * Math.signum(error))) : error;
+        error %= 360;
+        error = Utils.checkDeadband(error, 0.01);
+
+        angleMotor.set(
+                ControlMode.Position,
+                angleMotor.getSelectedSensorPosition() + angle_unitModel.toTicks(error)
+        );
+    }
+
+    /**
+     * Gets the velocity of the module.
+     *
+     * @return the velocity of the module. [m/s]
+     */
+    public double getVelocity() {
+        return drive_unitModel.toVelocity(driveMotor.getSelectedSensorVelocity());
+    }
+
+    /**
+     * Sets the velocity of the module.
+     *
+     * @param velocity is the velocity to set the module to. [m/s]
+     */
+    public void setVelocity(double velocity, double timeInterval) {
+        velocity = Utils.checkDeadband(velocity, 0.1);
+
+        linearSystemLoop.setNextR(VecBuilder.fill(velocity));
+        linearSystemLoop.correct(VecBuilder.fill(getVelocity()));
+        linearSystemLoop.predict(timeInterval);
+
+        driveMotor.set(ControlMode.PercentOutput, linearSystemLoop.getU(0) / NOMINAL_VOLTAGE);
```

Comment:
use `setVoltage` instead

Reply:
I wanted to ask really, why are you using the WPILib versions of the motor controllers?